packages/block-library/src/term-template/index.php

```diff
@@ -38,18 +37,25 @@ function render_block_core_term_template( $attributes, $content, $block ) {
 		'hide_empty' => $query['hideEmpty'] ?? true,
 	);
 
-	// We set parent to 0 only if we show all terms as hierarchical or we show top-level terms.
-	if ( ( 'all' === $terms_to_show && ! empty( $query['hierarchical'] ) ) || 'top-level' === $terms_to_show ) {
-		$query_args['parent'] = 0;
-	} elseif ( 'subterms' === $terms_to_show ) {
-		// Check if we're in a taxonomy archive context.
-		if ( is_tax( $query_args['taxonomy'] ) ) {
-			// Get the current term ID from the queried object.
-			$current_term_id = get_queried_object_id();
-			if ( $current_term_id && $current_term_id > 0 ) {
-				$query_args['parent'] = $current_term_id;
-			}
+	// We set parent only when inheriting from the taxonomy archive context or not
+	// showing nested terms, otherwise nested terms are not displayed.
+	if (
+		isset( $query['inherit'] )
```

Comment:
This might be related to my `inherit` comments. There shouldn't be a value here right now..